packages/webpack/src/utils/postcss-v8.js

```diff
@@ -178,6 +178,22 @@ export default class PostcssConfig {
       }
     }
 
+    if (typeof this.postcssOptions === 'function') {
+      return {
+        postcssOptions: (loaderContext) => {
+          const postcssOptions = merge(
+            {},
+            this.defaultPostcssOptions,
+            this.postcssOptions(loaderContext)
+          )
+          this.loadPlugins(postcssOptions)
+
```

Comment:
It seems we are losing part of features comparted to non function mode  (`preset`, `execute`) (and it might be source of more inconsistent behavior in the future)

Can we maybe convert L200:221 (`isPureObject(postcssOptions))`) into an inline function to immediately call and return or delay to use by function wrapper to get context?

Reply:
.. we could wrap some parts in a function wrapper, but the main "problem" here is that preset and execute needs to be evaluated "at beginning" and not when the function itself is executed. The only return type the function can give is the "postcssOptions" itself as an object.
Therefore I would propose to allow a config like this:
postcss: {
  execute: true,
  order: 'presetEnvAndCssnanoLast',
  preset: {
    stage: 2
  },
  postcssOptions: // config or function.. well of course old plugins can still be used, but to keep it easier I would suggest also allowing setting configs inside this object too.
}

I will commit my idea, let me know what you think of it

Reply:
Seems a nice idea just maybe we can use `postcss.options` for new options passing/function possibility?

Reply:
> @simllll Since linked issue is related to postcss < 8 (current) shall we support same feature in `postcss.js` util?

Could be possible, but it needs postcss-loader >= 4.x, see release notes https://github.com/webpack-contrib/postcss-loader/releases/tag/v4.0.0
and I couldn't run nuxt with postcss-loader 4.x without postcss8. Therefore I guess it's not really necessary.

Reply:
> Seems a nice idea just maybe we can use `postcss.options` or `postcss.postcssOptions` for new options passing/function possibility? (avoid two top level options for postcss)

Not quite sure what you mean, is `postcssOptions` good or should I go with `options` instead?
I just finished some small tests, looks good to me. Fully backwards compatible and no duplicate code anymore :)